File: leetcode-python-go/meituan-011. 搭配出售/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

var a, b, c, d, e, f, g int

func main() {
	in := bufio.NewReader(os.Stdin)
	fmt.Fscan(in, &a, &b, &c, &d, &e, &f, &g)
	var res int
	var x int
	for d > 0 && (a > 0 || b > 0 || c > 0) {
		if a > 0 && e >= f && e >= g {
			x = min(a, d)
			res += e * x
			a -= x
			d -= x
			e = -1
		} else if b > 0 && f >= e && f >= g {
			x = min(b, d)
			res += f * x
			b -= x
			d -= x
			f = -1
		} else if c > 0 && g >= e && g >= f {
			x = min(c, d)
			res += g * x
			c -= x
			d -= x
			g = -1
		}
	}
	fmt.Println(res)
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}

File: leetcode-python-go/meituan-011. 搭配出售/main.py
def main():
    res = 0
    a, b, c, d, e, f, g = map(int, input().split())
    if e < f:
        a, b, e, f = b, a, f, e
    if e < g:
        a, c, e, g = c, a, g, e
    if f < g:
        b, c, f, g = c, b, g, f
    if a >= d:
        return d * e
    res, d = res + a * e, d - a
    if b >= d:
        return res + d * f
    res, d = res + b * f, d - b
    return res + min(c, d) * g

print(main())
